src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -135,10 +136,11 @@ internal static class ParameterHelpers
 
                 if (mustBeLastParameter == null)
```

Comment:
>if (mustBeLastParameter == null) [](start = 16, length = 32)

It looks like we can safely move this ```if``` statement right after the code that reports ERR_ThisInBadContext. This way all checks specific to ```ParameterSyntax``` will be grouped together and can be placed under umbrella of a single type test.